File: Server_Spring/Elderly_Care/src/main/java/kr/spring/care/caregiver_page/service/CaregiverPageService.java
package kr.spring.care.caregiver_page.service;


import java.util.List;

import kr.spring.care.admin.DTO.CaregiverDTO;
import kr.spring.care.senior_page.dto.MatchingDTO;
import kr.spring.care.user.entity.User;
import kr.spring.care.user_page.dto.UserDTO;

public interface CaregiverPageService {
	
	// 기본사항(User)
	public UserDTO myInfo(long userId);
	
	// 특정사항(Senior) 정보
	public CaregiverDTO caregiverInfo(long userId);
	
	// 회원정보 수정(기본+특정)
	public void editUser(UserDTO userDTO);
	
	// 비밀번호 수정
	public void editPw(User user);
	
	// 매칭 정보
	public List<MatchingDTO> matchingInfo(long userId);
	
}

File: Client_Android/app/src/main/java/com/example/elderlycare/mypage/service/CaregiverPageService.kt
package com.example.elderlycare.mypage.service

import com.example.elderlycare.mypage.vo.CaregiverDTO
import com.example.elderlycare.mypage.vo.MatchingDTO
import com.example.elderlycare.mypage.vo.MatchingResponse
import com.example.elderlycare.mypage.vo.SeniorDTO
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface CaregiverPageService {

    @GET("caregiverInfo/{userId}")
    fun caregiverInfo(@Path("userId") userId: Long): Call<CaregiverDTO>

    @GET("matchingInfo/{userId}")
    fun matchingInfo(@Path("userId") userId: Long): Call<MatchingResponse>

}